File: app/src/main/java/com/guuguo/gank/db/converter/RoomDataConverter.java
package com.guuguo.gank.db.converter;

import android.arch.persistence.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import org.w3c.dom.Comment;

import java.util.Date;
import java.util.List;


public class RoomDataConverter {
    @TypeConverter
    public static Date toDate(Long timestamp) {
        return timestamp == null ? null : new Date(timestamp);
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

    @TypeConverter
    public static List<String> toStringList(String json) {
        return new Gson().fromJson(json, new TypeToken<List<String>>() {
        }.getType());
    }

    @TypeConverter
    public static String toStringListJson(List<String> list) {
        return new Gson().toJson(list);
    }


}

File: app/src/main/java/com/guuguo/gank/db/converter/RoomDataConverter.kt
package com.guuguo.gank.db.converter

import androidx.room.TypeConverter
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.util.*


class RoomDataConverter {
    @TypeConverter
    fun toDate(timestamp: Long?): Date? {
        return if (timestamp == null) null else Date(timestamp)
    }

    @TypeConverter
    fun toTimestamp(date: Date?): Long? {
        return date?.time
    }

    @TypeConverter
    fun toStringList(json: String): List<String> {
        return Gson().fromJson(json, object : TypeToken<List<String>>() {
        }.type)
    }

    @TypeConverter
    fun toStringListJson(list: List<String>): String {
        return Gson().toJson(list)
    }


}
